### PR TITLE
Add RTCRtpParameters.codec, RTCRtpCodecParameters, and RTCEncodingPar…

### PR DESCRIPTION
**Pull request by pthatcherg**
List discussion needed.

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -2530,23 +2539,55 @@
...
+            <p>The RTP payload type.  This value can be set in the
+            RTCRtpParameters.encodings[0].payloadType to control which
+            codec should be used to send a given encoding.
+            </p>
+          </dd>
+
+          <dt>DOMString name</dt>
+
+          <dd>
+            <p>The name of the codec, such as "opus", "vp8", and "h264".</p>
```
Comment (pthatcherg) [inline webrtc.html]: FYI, we can change this to match the results from the discussion on getCapabilities around mime type vs name.

Comment (martinthomson) [inline webrtc.html]: Codecs don't have names, they have media types.

File: webrtc.html [hunk 2]
```diff
@@ -2530,23 +2539,55 @@
...
           <dd>
             <p>
-              Indicates the relative priority of this encoding, across
-              all RtpSenders of a given PeerConnection.  When there is
+              The relative priority of this encoding across all
+              RtpSenders of a given PeerConnection.  When there is
               limited bandwidth available to a PeerConnection, higher
               prioirty encodings will be sent with more bandwidth, and
               lower priority encodings will be sent with less
               bandwidth.
```
Comment (martinthomson) [inline webrtc.html]: This isn't enough information to implement consistent behaviour.

Comment (pthatcherg) [inline webrtc.html]: This is being updated in https://github.com/w3c/webrtc-pc/pull/259.  You may like that version better.

And that really has nothing to do with this PR.  It looks like I accidentally changed some grammar/formatting.

File: webrtc.html (hunk 1 above)
Comment (pthatcherg) [inline webrtc.html]: Like I said, I'll just change this to match what is decided in getCapabilities.

Comment (pthatcherg) [inline webrtc.html]: OK, it's been changed to mimeType to match https://github.com/w3c/webrtc-pc/pull/269.

At Harald's request, I also added clockRate and sdpFtmpLine.

**Comment by martinthomson** [comment]
This is an odd control surface.  If you take the view that these are domain objects, then you could represent them with interfaces.  I think that at this point, it would make a lot of sense to take a step back and try to model this out with interfaces.  Then you could use actual objects as the basis of the selection, rather than this odd payload type indirection.

**Comment by alvestrand** [comment]
As noted in Seattle:
The PR needs to include info about what happens at renegotiation - whether payload type goes back to "browser selects", PT gets kept no matter if it works or not, or whether it's "browser, do something sensible".

**Comment by alvestrand** [comment]
From Seattle:
Another approach in this PR is to make the "codecs" sequence updateable - allow reorder and deletion, but not adding new ones or change existing ones.

**Comment by pthatcherg** [comment]
Here is that approach: https://github.com/w3c/webrtc-pc/pull/298/f

On Wed, Sep 9, 2015 at 10:45 AM, Harald Alvestrand <notifications@github.com

**Comment by alvestrand** [comment]
Closing per Seattle discussions, obsoleted by #298.

**Pull request closed without merging**